web/thesauruses/javascript/ECMAScript 2009/strings.json

```diff
@@ -0,0 +1,344 @@
+        "code": [
+            "const str = \"Hello World\";",
+            "const str2 = str;"
+        ]
+      },
+      "destroy_string": {
+        "not-implemented": true,
+        "name": "Destroy string"
+      },
+      "length_of_string": {
+        "name": "Length of string",
+        "code": "str.length"
+      },
+      "max_length_of_string": {
+        "name": "Maximum length of string",
+        "code": "str.length = 5;"
+      },
+      "clear_string": {
+        "name": "Clear/empty string",
+        "code": [
+            "str = \"\";",
+            "str.length = 0;"
+        ]
+      },
+      "is_empty": {
+        "name": "Is string empty?",
+        "code": [
+            "str.length === 0",
+            "str === \"\""
+        ]
+      },
+      "concatenate_two_strings": {
+        "name": "Concatenate/join two strings",
+        "code": [
+            "str + str2",
+            "str.concat(str2)"
+        ]
+      },
+      "concatenate_many_strings": {
+        "name": "Concatenate/join many strings",
+        "code": [
+            "str + str2 + str3 + str4",
+            "str.concat(str2, str3, str4)"
+        ]
+      },
+      "is_all_alphabetical": {
+        "name": "Is string all alphabetical characters?",
+        "code": "(/^[a-zA-Z]+$/g).test(str)"
+      },
+      "is_all_numerical": {
+        "name": "Is string all numerical characters?",
+        "code": "(/^[0-9]+$/g).test(str)"
+      },
+      "is_all_alphanumeric": {
+        "name": "Is string all alphanumeric characters?",
+        "code": "(/^[a-zA-Z0-9]+$/g).test(str)"
+      },
+      "is_decimal": {
+        "name": "Is string a decimal number?",
+        "code": "(/^[0-9]+\\.[0-9]+$/g).test(str)"
+      },
+      "is_all_whitespaces": {
+        "name": "Is string all whitespace characters?",
+        "code": "(/^[\\s]+$/g).test(str)"
+      },
+      "is_all_uppercase": {
+        "name": "Is string all uppercase characters?",
+        "code": "str === str.toUpperCase()"
+      },
+      "is_all_lowercase": {
+        "name": "Is string all lowercase characters?",
+        "code": "str === str.toLowerCase()"
+      },
+      "is_in_titlecase": {
+        "name": "Is string formatted in title case?",
+        "code": "str === str.replace(/\\w+/g, (w) => w[0].toUpperCase() + w.slice(1).toLowerCase())"
+      },
+      "does_substring_exist": {
+        "name": "Does a substring exist in a string?",
+        "code": [
+            "str.indexOf(substring) !== -1"
+        ]
+      },
+      "find_start_index_of_substring": {
+        "name": "Find index of where a substring starts",
+        "code": [
+            "str.indexOf(substring)",
+            "str.search(substring)"
+        ]
+      },
+      "find_start_index_of_additional_substring": {
+        "name": "Find index of an additional substring (or starting at another index)",
+        "code": [
+            "str.indexOf(substring, index)",
+            "str.search(substring)"
```

Comment:
`search()` only finds the first instance of it. You could write a regex to find others, but that's really the `match()` function. I think just keep the `indexOf()`

(Similar for other `search()` below.)